### PR TITLE
Dockerize Parser service

### LINKED ISSUE #2111: Write tests and enable Parser service
We landed the Parser service today, but we have it not starting the actual parser, nor do we have any tests. We need to fix both things.  @c3ho added Jest configs for both unit and e2e tests, but there aren't any.

You can get some idea of how to work across services in tests by looking at my auth test [here](https://github.com/Seneca-CDOT/telescope/blob/master/src/api/auth/test/e2e/auth-flows.test.js).

### PR DESCRIPTION
**Pull request by TueeNguyen**
## Issue This PR Addresses
Fixes #2111 

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [ ] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description
- Revert all commented parser's docker code
- Add code to fit in with recent docker changes
- Rework parser's docker file to better practice

## Steps to test the PR
- `pnpm services:start`
- go to `localhost:8000` AND THAT'S ALL YOU NEED, NO `pnpm start`

## Checklist

- [ ] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

**Comment by TueeNguyen** [comment]
Wait for #3040 to merge first

File: docker/production.yml [hunk 1]
```diff
@@ -100,10 +100,10 @@ services:
     image: docker.cdot.systems/posts:${DOCKER_DEPLOY_TAG:-latest}
     restart: unless-stopped
 
-  # TODO: https://github.com/Seneca-CDOT/telescope/issues/2111
   # parser service
-  # parser:
-  #   restart: unless-stopped
+  parser:
+    image: docker.cdot.systems/posts:${DOCKER_DEPLOY_TAG:-latest}
```
Comment (TDDR) [inline docker/production.yml:105]: should be `parser` instead of `posts`

**Review by TDDR — CHANGES_REQUESTED** [review]

File: src/api/parser/Dockerfile [hunk 2]
```diff
@@ -1,13 +1,34 @@
...
+## Deploy ########################################################################
+# Stage for running our app
+FROM base as deploy
+
 COPY --chown=node:node . .
 
-RUN npm install --only=production --no-package-lock
+COPY --chown=node:node --from=dependencies /app/node_modules ./node_modules
+
+ENV PORT=10000
```
Comment (Kevan-Y) [inline src/api/parser/Dockerfile:26]: I think we can replace to `PARSER_PORT`
```suggestion
ENV PARSER_PORT=10000
```

**Review by Kevan-Y — CHANGES_REQUESTED** [review]

**Comment by Kevan-Y** [comment]
Image size compare to before
```bash 
❯ docker images | grep refactor/parser
refactor/parser                                 old                de119410c427   14 seconds ago       185MB
refactor/parser                                 new                5efd1346e57b   About a minute ago   170MB
```

**Comment by TueeNguyen** [comment]
> Image size compare to before
> 
> ```shell
> ❯ docker images | grep refactor/parser
> refactor/parser                                 old                de119410c427   14 seconds ago       185MB
> refactor/parser                                 new                5efd1346e57b   About a minute ago   170MB
> ```

Small but good improvement

File: docker/development.yml [hunk 3]
```diff
@@ -85,11 +85,17 @@ services:
...
-  #   environment:
-  #     # For development and testing, the Parser service needs to contact the users
-  #     # service directly via Docker vs through the http://localhost domain.
+  parser:
+    build:
+      context: ../src/api/parser
+      cache_from:
+        - docker.cdot.systems/parser:buildcache
+    ports:
+      - '10000:10000'
```
Comment (humphd) [inline docker/development.yml:94]: Why do we need a port open?  It's accessible via Traefik.

File: docker/gitpod.yml [hunk 4]
```diff
@@ -71,8 +71,10 @@ services:
...
-  # parser:
-  #   environment:
-  #     # For development and testing, the Parser service needs to contact the users
-  #     # service directly via Docker vs through the http://localhost domain.
+  parser:
+    build:
+      context: ../src/api/parser
+      cache_from:
+        - docker.cdot.systems/parser:buildcache
+    ports:
```
Comment (humphd) [inline docker/gitpod.yml:79]: Same question here.

File: src/api/parser/Dockerfile [hunk 5]
```diff
@@ -1,13 +1,33 @@
...
+COPY --chown=node:node --from=dependencies /app/node_modules ./node_modules
+
+ENV PARSER_PORT=10000
 
 USER node
 
-CMD ["dumb-init", "node", "src/server.js"]
+# Docker Healthcheck
+HEALTHCHECK --interval=30s --timeout=30s --start-period=5s --retries=3 \
+	CMD wget --no-verbose --tries=1 --spider localhost:${PORT}/healthcheck || exit 1
```
Comment (humphd) [inline src/api/parser/Dockerfile:31]: `PORT` should be `PARSER_PORT`

**Review by humphd — CHANGES_REQUESTED** [review]

**Review by humphd — APPROVED** [review]

**Pull request merged**